src/model/document.ts

```diff
@@ -138,6 +138,10 @@ export class Document implements Feature, Queryable {
     this.warnings = Array.from(base.warnings);
   }
 
+  static isDocument(value: any): value is Document {
+    return typeof value === 'object' && !!value.constructor.isDocument;
```

Comment:
Your fix is good and you should feel good.

*Why did I think null would have a constructor just because it is an object?*